pkg/storage/watch_cache_test.go

```diff
@@ -230,6 +231,25 @@ func TestEvents(t *testing.T) {
 	}
 }
 
+func TestWaitUntilFreshAndList(t *testing.T) {
+	store := newWatchCache(3)
+
+	// In background, wait some time and then update the store.
```

Comment:
Can we do that (without polluting production code with some callbacks or sth like that)? I would like to avoid that.